state/azure/tablestorage/tablestorage.go

```diff
@@ -122,7 +127,12 @@ func (r *StateStore) Get(req *state.GetRequest) (*state.GetResponse, error) {
 func (r *StateStore) Set(req *state.SetRequest) error {
 	r.logger.Debugf("saving %s", req.Key)
 
-	return r.writeRow(req)
+	err := r.writeRow(req)
+	if req.ETag != "" {
+		return state.NewETagError(state.ETagMismatch, err)
+	}
```

Comment:
@yaron2 - can you explain how this works? I'm not familiar with those code but it looks like this would return `state.NewETagError` every time an ETag is specified.

Reply:
Theres an err check missing here, I'll push a fix later